modules/apps/data-engine/data-engine-rest-test/src/testIntegration/java/com/liferay/data/engine/rest/resource/v1_0/test/DataDefinitionResourceTest.java

```diff
@@ -212,12 +214,17 @@ private void _testGetSiteDataDefinitionsPage(
 			dataDefinitionResource.getSiteDataDefinitionsPage(
 				siteId, keywords, Pagination.of(1, 2), null);
 
-		Assert.assertEquals(1, page.getTotalCount());
+		IdempotentRetryAssert.retryAssert(
```

Comment:
@jeyvison @marcelabc what's the problem you are trying to solve with IdempotentRetryAssert here?
Once the Page is returned, it is not going to change anymore, no matter how many times you try to call its getTotalCount() and getItems(), you always get the same values.

Do you mean to retry on the "dataDefinitionResource.getSiteDataDefinitionsPage()"?

And I run your tests without IdempotentRetryAssert, it still passes.

So please clarify what's the problem you are trying to solve first. But anyway, this usage of IdempotentRetryAssert is not doing anything meaningful.

Reply:
@marcelabc @gabrielwas Can you answer Shuyang please? Thanks :)

Reply:
@shuyangzhou you're right. The IdempotentRetryAssert should call "dataDefinitionResource.getSiteDataDefinitionsPage()" instead of "page.getTotalCount()". You run without IdempotentRetryAssert, and it passed because it is an intermittent fault, due to the indexing process of ElasticSearch. It uses to fail when running ci tests.
If you check the "testGetSiteDataDefinitionsPage" method, it calls "_testGetSiteDataDefinitionsPage" method a few times in sequence, and inside it, we save and delete DataLayouts. Then, we guess sometimes there is not enough time to reindex it. That's why we are trying to fix this with the retryAssert.
Do you have any suggestion?

Reply:
Then this is really a search question. When running in "test" model, search is supposed to behave in a sync mode.

We set CI to test model by https://github.com/liferay/liferay-portal/blob/master/build-test.xml#L8821
And search is reading this property in places like https://github.com/liferay/liferay-portal/blob/master/modules/apps/portal-search-elasticsearch6/portal-search-elasticsearch6-impl/src/main/java/com/liferay/portal/search/elasticsearch6/internal/ElasticsearchIndexWriter.java#L76

Just search for "PortalRunMode.isTestMode()" calls.

If you see under test model, search is still behaving asynchronously, then it is a search issue, @arboliveira can help you with that.